game.js

```diff
@@ -31,11 +31,44 @@ class Game {
 
         this.log("Connecting to game.");
 
-        // TODO: Command line options to allow undo?
-        //
-        this.socket.on('game/' + game_id + '/undo_requested', (undodata) => {
-            this.log("Undo requested", JSON.stringify(undodata, null, 4));
-        });
+        if (config.UNDO && !config.UNDORANKED && !config.UNDOUNRANKED) {
+            this.socket.on('game/' + game_id + '/undo_requested_in_any_type_of_game', (undodata) => {
+                this.log("Undo requested in any type of game", JSON.stringify(undodata, null, 4));
+                // allow the undo and process it ..., then
```

Comment:
This should be something like:

```
this.conn.emit('game/undo/accept', this.auth({
     game_id: this.game_id,
     move_number: undodata,
}));
```

Reply:
feel free to commit on top of my PR and submit it @windo, i'd be glad :+1: 

i dont know anything about the connection things, so i currently cant help any further :+1:

Reply:
I believe just this is all you need to get the bot to accept the undo. Then you'd also need to kill the bot I think. So maybe the `emit` from above, followed by an `this.ensureBotKilled()` and a maybe a `this.scheduleRetry()`. You might not need a retry technically because it would be the opponents move and we should get a new gamedata when they make a move anyway. Not sure.

But the PR should probably do something that is at least approximately capable of handling undos, not much reason to merge a configuration for something that is not supported, I think.

Reply:
i really cant do that easily, so if you or someone else `git pull upstream undo-undoranked-undounranked-v1` i think i would really like it

but whenever one wants/can of course

as for me, it's really all i can do on support, for now at least (with my current skills and understanding